### PR TITLE
Backport PR #45287 on branch 1.4.x (BUG: frame[x].loc[y] inconsistent with frame.at[x, y]) (#45287)

### PR DESCRIPTION
**Pull request by phofl**
Backport PR #45287, see https://github.com/pandas-dev/pandas/issues/47867#issuecomment-1198412271

Lets see if this works

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
don't worry about the mypy failures. imo it is main that is wrong since we have traditionally always run the typing validation against the latest numpy release, which is what's happening here.

**Comment by phofl** [comment]
Yep thx

File: doc/source/whatsnew/v1.4.4.rst [hunk 1]
```diff
@@ -16,6 +16,7 @@ Fixed regressions
 ~~~~~~~~~~~~~~~~~
 - Fixed regression in :func:`concat` materializing :class:`Index` during sorting even if :class:`Index` was already sorted (:issue:`47501`)
 - Fixed regression in setting ``None`` or non-string value into a ``string``-dtype Series using a mask (:issue:`47628`)
+- Fixed regression in :meth:`loc.__getitem__` with a list of keys causing an internal inconsistency that could lead to a disconnect between ``frame.at[x, y]`` vs ``frame[y].loc[x]`` (:issue:`22372`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.4.4.rst:19]: I would probably move this to bug fix and put reference to the new issue in the regression section

Comment (phofl) [reply doc/source/whatsnew/v1.4.4.rst]: Done

Comment (simonjayhawkins) [reply doc/source/whatsnew/v1.4.4.rst]: I would also add a release note for the #47867 issue in the regression section (in the addition to the bugfix release note)

or are you doing this when you add the test?

Comment (phofl) [reply doc/source/whatsnew/v1.4.4.rst]: Wanted to do this with the test, but no strong preference

Comment (simonjayhawkins) [reply doc/source/whatsnew/v1.4.4.rst]: sure. merge when ready.

**Comment by phofl** [comment]
Failure unrelated, see #47945

**Pull request merged**